Restructure for hugo

ralphbean opened the pull request.

ralphbean committed:
Restructure for hugo

ralphbean commented:
I'm seeing really weird failures here with hugo that I don't understand.

The `theme` is currently missing. Clone it into the themes directory first.

Then, run `hugo server` to build and server locally.

Notice that these urls work:

* http://localhost:1313/decisions/ - the list of adrs
* http://localhost:1313/decisions/0000-adr-template/ - individual adrs
* http://localhost:1313/architecture/image-controller/ - individual design docs

But this url does _not_ work:

* http://localhost:1313/architecture/ - the list of design docs.

That last one should be an index of the design documents... but, it is empty. I can make it render with a list of design docs if I _remove_ what seems to be an arbitrary number of the _ADRs_ in the `decisions/` dir, which just makes no sense at all.

zregvart committed:
Use Hugo modules to fetch the theme

zregvart committed:
Rename _index.md to index.md

avi-biton commented on hugo.toml:
```diff
@@ -0,0 +1,42 @@
+
+    # Style options: default (light-blue), blue, green, marsala, pink, red, turquoise, violet
+    style = "default"
+
```
Try adding these lines under [params] to be able to open 
http://localhost:1313/architecture/
and  http://localhost:1313/decisions

```
    mainSections = ["architecture", "decisions"]
    defaultDescription = "Site template made by Devcows using Hugo"
```

zregvart commented:
Also have a look at #184

ralphbean commented:
> Do we need the the HAS documentation?
@avi-biton let's keep it in for the restructuring. Removing it should be documented as a decision on its own.

ralphbean committed:
Get both sections to render by specifying mainSections

ralphbean commented:
Now that the architecture repo is building successfully at https://konflux-ci.dev/architecture/ with jekyll, I'm not sure I have the motivation to finish this PR - there's still a good bit more cleanup to do to get the style to match https://konflux-ci.dev/

Do either of you want to pick up the remaining work in this PR? If so, feel free to, otherwise I think I'll drop it.

zregvart commented:
> Do either of you want to pick up the remaining work in this PR? If so, feel free to, otherwise I think I'll drop it.

Beside my bias towards Hugo, I'm not sure much is gained by transition from Jekyll to Hugo. Is there something we wish to gain with this?

ralphbean commented:
@zregvart I think the only thing at this point is alignment of the _style_ between https://konflux-ci.dev/ and https://konflux-ci.dev/architecture/

zregvart commented:
In https://github.com/konflux-ci/konflux-ci.github.io/pull/39 the custom CSS and images are pulled in to a theme that can be addressable via Hugo modules, i.e. changing the `theme` in `hugo.toml` something like:

```diff
diff --git a/hugo.toml b/hugo.toml
index b950f08..8172d71 100644
--- a/hugo.toml
+++ b/hugo.toml
@@ -1,7 +1,7 @@
 baseURL = 'http://localhost:1313/'
 languageCode = 'en-us'
 title = 'Standalone Konflux Architecture Docs'
-theme = 'github.com/devcows/hugo-universal-theme'
+theme = ['github.com/konflux-ci/konflux-ci.github.io/website/themes/konflux', 'github.com/devcows/hugo-universal-theme']
 
 # Site language. Available translations in the theme's `/i18n` directory.
 defaultContentLanguage = "en"

```

Would bring in the styling of [konflux-ci.dev](https://konflux-ci.dev/)

zregvart committed:
Inherit theme from konflux-ci.dev

This adds the theme from konflux-ci.dev as a Hugo module. Any styling
changes there are applied to this website as well.
The version is pinned, so updates to konflux-ci.dev will not be applied
without running `hugo mod get -u`.

zregvart commented:
With #188 the overall styling of konflux-ci.dev website can be inherited here as well.

zregvart committed:
Custom layout and front matter for ADR decisions

Adds custom "list" layout and modifies the markdown files to add front
matter.

zregvart commented:
In #189 I've restructured the ADR decisions to add front matter and added a custom list layout for decisions, it renders a bit nicer now.

ralphbean commented:
@zregvart 00d755c66a03e90ac5c3be40da7d0d9147c83a3c had a side effect that I don't understand. Before it, urls like http://localhost:1313/architecture/release-service/ would render correctly, but after it they do not.

zregvart committed:
Fix architecture section rendering

zregvart committed:
Generate API reference using Make

This was helpful to generate the API docs locally so I can preview and
run the link checker to check for broken links.

Adds the `tools/go.mod` so the version of `crd-ref-docs` can be pinned,
and the tool can be run using `go run`.

In essence one can do `make api-docs` to generate the Markdown files in
`content/ref`. There is a small issue with the
`service-provider-integration-operator`, it needs an additional
`go mod tidy` if golang version is newer than 1.20. To fix that run
`make api-docs` once to clone the repository, then
`cd crd-temp/service-provider && go mod tidy` and rerun `make api-docs`.

ralphbean committed:
Fix some links from the overview doc

zregvart commented:
@ralphbean in #190 I've used `relref` for links and tweaked a number of links so that they point to the local content rather than the published site URL (used to be https://redhat-appstudio.github.io/architecture/). The link checker I use checks relative links and fragments, the only broken links with this seem to be references to removed objects from the API docs.

zregvart commented:
#191 was useful for previewing locally

ralphbean committed:
Merge pull request #191 from zregvart/pr/api-docs-makefile

Generate API reference using Make
